browser/components/EditProject.tsx

```diff
@@ -0,0 +1,315 @@
+/* Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+import * as React from 'react';
+import SweetAlert from 'react-bootstrap-sweetalert';
+import Select from 'react-select';
+import { postJSON, reqJSON } from '../util/index';
+
+interface Props {
+  groupList: [
+    {
+      goal: string;
+      group_id: number;
+      group_name: string;
+      contribMTD: number;
+      contribMonth: number;
+      contribWeek: number;
+      contribYear: number;
+      numGroups: number;
+      numUsers: number;
+      projects: number[];
+      sponsor: string;
+    }
+  ];
+  project: string;
+}
+
+interface State {
```

Comment:
Goodness, these types. You might have a more fun time breaking this apart and composing them on-the-fly with `&` and `|`: https://www.typescriptlang.org/docs/handbook/advanced-types.html#intersection-types

Reply:
Interesting. I will read up and may do a sweep through later. Thanks for the info.